go/libraries/doltcore/row/keyless_row.go

```diff
@@ -264,3 +264,52 @@ func (r keylessRow) TaggedValues() (TaggedValues, error) {
 func (r keylessRow) Format() *types.NomsBinFormat {
 	return r.val.Format()
 }
+
+// ReduceToIndexKeys creates a full key, a partial key, and a cardinality value from the given row
+// (first tuple being the full key). Please refer to the note in the index editor for more information
+// regarding partial keys.
+func (r keylessRow) ReduceToIndexKeys(idx schema.Index) (types.Tuple, types.Tuple, types.Tuple, error) {
```

Comment:
cool if I punt on this for now? This will need to be rewritten w/ taggedValues as the main interface for extracting indexes. I also think keyed rows being explicit about an empty value tuple is clear and correcy, I wasted a day on a strange bug b/c `types.Tuple{}` is not nbf compatible, but `types.EmptyTuple()` is.

Reply:
It's fine to punt on `ReduceToIndexKeys`, and thinking about it more it's fine that keyed rows are explicit in returning an empty value. If possible I'd still like to have the different interfaces for them both, even if it's through the implementation of a private function to mark the explicit distinction between the two.